### DISCUSSION TITLE
How often does tuh_task() need to be called?

### DISCUSSION THREAD

**Discussion by mdrobot7** [hathach/tinyusb#1972] [Q&A] [answered]
(Referencing the getting started docs page)

How often does the `tuh_task()` function need to be called? What happens if `your_application_code()` takes longer than a few cycles (a few milliseconds, for example)? Are there any negative effects?

Why is it recommended to run `tuh_task()` in a `while(true)` rather than run it in an interrupt handler every `n` microseconds/cycles? Is it only to prevent delays in handling other interrupts while a theoretical `tuh_task()` interrupt would be running?

**Comment by hathach** [comment, marked as answer]
it should be called when there is an usb event in the usbh queue i.e something happened in the bus
